gax-java/gax/src/main/java/com/google/api/gax/rpc/Watchdog.java

```diff
@@ -220,6 +220,16 @@ public void onStartImpl(StreamController controller) {
             public void disableAutoInboundFlowControl() {
               Preconditions.checkState(
                   !hasStarted, "Can't disable automatic flow control after the stream has started");
+
+              synchronized (lock) {
+                // WatchdogStream is created with waiting state. If auto flow control is disabled,
+                // set the state to be IDLE, which will be updated to WAITING when onRequest() is
+                // called.
+                if (state == State.WAITING) {
+                  state = State.IDLE;
+                }
```

Comment:
I dont think you need the conditional either. The state transition for a response observer is strictly: constructed -> start -> everything else
So you should just blindly overwrite it here